docs/contracts.md

```diff
@@ -270,3 +412,26 @@ fn get_receive_sender(start: *mut u8);
 /// 32 bytes.
 fn get_receive_owner(start: *mut u8);
 ```
+
+## Fallback entrypoint handling
+
+A V1 contract may have a fallback entrypoint. This is by definition the
+entrypoint with the empty name, e.g., for a contract named `foo` the fallback
+entrypoint has a receive name `foo.`. The semantics of fallback entrypoints is
+that if a method on a contract is invoked, if there is no exact match for an
+entrypoint, then the fallback entrypoint is invoked, if it exists.
+
+Since this means that the executed entrypoint may be different from the one that
+was named in, e.g., transaction, it is useful to expose the named entrypoint in
+the contract. That is the purpose of the following two methods which are only
+allowed for receive methods. Invoking these in an init method will lead to a
+runtime error and contract termination.
+
+```rust
+/// Get the size of the entrypoint that was named.
+fn get_receive_entrypoint_size() -> u32;
+
+/// Write the receive entrypoint name into the given location.
+/// It is assumed that the location contains enough space to write the name.
+fn get_receive_entrypoint(start: *mut u8);
```

Comment:
I also struggled when making this choice due to the inconsistency, but ultimately it is a better choice than following the state entry read, because there is not any value in reading a partial entrypoint name.